src/server/shell/decoration/basic_decoration.cpp

```diff
@@ -266,9 +266,15 @@ void msd::BasicDecoration::set_cursor(std::string const& cursor_image_name)
     shell->modify_surface(session, decoration_surface, spec);
 }
 
+void msd::BasicDecoration::set_scale(float scale)
+{
+    scale_ = scale;
+    window_state_updated();
+}
```

Comment:
Is access to `scale_` threadsafe? AFAICS it isn't atomic, guarded by a mutex, or guaranteed to be accessed on a single thread